### ISSUE TITLE
setupgeom is working?

### ISSUE THREAD

**Issue by mohammadamirb** [dftbplus/dftbplus#1074]
**Bug discription**
I have did some tansport calculations using setupgeom tool of the older version of DFTB+. But this tool seems to have problem in newer ones as it cant reorder atoms and stops with an error. 
 _ERROR!
-> atom 0               is found in more than one region
ERROR STOP_ 
However, release 20.2.1 works fine!

**To Reproduce**
1. Use precompiled version 21.1 or 22.1 setupgeom tool to see what happened.

The error is : 
...
```
***  Parsing and initializing

Parser version: 6

Interpreting input file 'setup_in.hsd'
--------------------------------------------------------------------------------
Reading SK-files:
 ../../../pbc-0-3/C-C.skf
 ../../../pbc-0-3/C-H.skf
 ../../../pbc-0-3/H-H.skf
Done.

 Maximum SK cutoff:   6.0220370936199998      (A)
ERROR!
-> atom 0               is found in more than one region
ERROR STOP 

Error termination. Backtrace:
#0  0x4197ce in __dftbp_common_globalenv_MOD_abortprogram
	at /mnt/scratch/aradi/distrib/dftb+/src/dftbp/common/globalenv.f90:136
#1  0x40cd26 in assigndeviceatoms
	at /mnt/scratch/aradi/distrib/dftb+/app/transporttools/helpsetupgeom.f90:120
#2  0x40cd26 in __transporttools_helpsetupgeom_MOD_setupgeometry
	at /mnt/scratch/aradi/distrib/dftb+/app/transporttools/helpsetupgeom.f90:76
#3  0x41712a in readtransportgeometry
	at /mnt/scratch/aradi/distrib/dftb+/app/transporttools/parser.f90:279
#4  0x41712a in __transporttools_parser_MOD_parsehsdinput
	at /mnt/scratch/aradi/distrib/dftb+/app/transporttools/parser.f90:121
#5  0x4012ec in setupgeom
	at /mnt/scratch/aradi/distrib/dftb+/app/transporttools/setupgeom.f90:24
#6  0x4012ec in main
	at /mnt/scratch/aradi/distrib/dftb+/app/transporttools/setupgeom.f90:10
```

Input and output files which, version 20.2.1 can run them are attached.
[setupgeomissue.zip](https://github.com/dftbplus/dftbplus/files/9228911/setupgeomissue.zip)

**Additional context**
This problem is also in the Manual, where the defenition of contacts blocks can not be interpreted by the tool as described in Manual, I mean "SpecifiedPLs = 2" (Page 149).

**Comment by bhourahine** [comment]
@pecchia This is still broken in the codebase.

**Comment by aradi** [comment]
In theory, it should have fixed in 66e1c5fc on the main branch. Let us know, if it does not work for you.
